### PR TITLE
Fix timeseries widgets wonkiness

### LINKED ISSUE #4751: New time granularity filter acting weird
This is on Chrome on our stats instance. No JS errors.

The new time granularity dropdown on time series is behaving sort of oddly. I find I have to make a selection twice before it seems to take effect. Or I'll make a selection on the dropdown, and the query will appear to refresh but then nothing changes and the dropdown reverts to the previous state.

[Here's a video of the behavior](http://recordit.co/2flIiumIbR). Notice how after I change it to `Week`, it gets stuck when I try to change it back to `Day`, etc.

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #4751

### CODE REVIEW

**Comment by tlrobinson** [comment]
I thought about trying to write a test for this, but I'm not sure the best way since the issue was with the the way the React component called the Redux actions (specifically it called `setDatasetQuery` which is async, then immediately `runQuery`, so it was re-running the old query).

File: frontend/src/metabase/qb/components/TimeseriesGroupingWidget.jsx [hunk 1]
```diff
@@ -58,11 +61,13 @@ export default class TimeseriesGroupingWidget extends Component<*, Props, *> {
                                     breakout
                                 );
                                 // $FlowFixMe
-                                setDatasetQuery({
-                                    ...card.dataset_query,
-                                    query
-                                });
-                                runQuery();
```
Comment (salsakran) [inline frontend/src/metabase/qb/components/TimeseriesGroupingWidget.jsx]: out of curiosity, what was going on here?

Comment (salsakran) [inline frontend/src/metabase/qb/components/TimeseriesGroupingWidget.jsx]: nvm.

Comment (tlrobinson) [inline frontend/src/metabase/qb/components/TimeseriesGroupingWidget.jsx]: Just left a comment on the PR.

**Review by salsakran — APPROVED** [review]
Code looks cleaner as I feel like setting+running query is better done as a single atomic function call.

Things seem to work AsExpected with the widget locally for me.

**Pull request merged**